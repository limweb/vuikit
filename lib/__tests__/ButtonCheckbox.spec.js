import { createVue, destroyVM, queryByTag, queryByTagAll, toJSON } from './_util'

describe('ButtonCheckbox', () => {
  let vm
  afterEach(() => {
    destroyVM(vm)
  })

  it('renders correctly', () => {
    vm = createVue(`
      <vk-button-checkbox :value="[10, 30]">
        <vk-button :value="10">Button 1</vk-button>
        <vk-button :value="20">Button 2</vk-button>
        <vk-button :value="30">Button 3</vk-button>
      </vk-button-checkbox>
    `)
    expect(toJSON(vm)).toMatchSnapshot()
  })

  it('triggers event change when a button was clicked', () => {
    const vm = createVue(`
      <vk-button-checkbox>
        <vk-button :value="10">Button 1</vk-button>
        <vk-button :value="20">Button 2</vk-button>
        <vk-button :value="30">Button 3</vk-button>
      </vk-button-checkbox>`
    )
    const checkbox = queryByTag(vm, 'vk-button-checkbox')
    const buttons = queryByTagAll(vm, 'vk-button')
    const cb = jest.fn()

    checkbox.$on('change', cb)
    buttons[0].$el.click()
    expect(cb).toHaveBeenCalled()
    destroyVM(vm)
  })
})
