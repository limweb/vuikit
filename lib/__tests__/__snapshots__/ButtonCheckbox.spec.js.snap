exports[`ButtonCheckbox renders correctly 1`] = `
Object {
  "child": Array [
    Object {
      "attr": Object {
        "class": "uk-button-group",
      },
      "child": Array [
        Object {
          "attr": Object {
            "class": Array [
              "uk-button",
              "uk-active",
            ],
            "type": "button",
          },
          "child": Array [
            Object {
              "node": "text",
              "text": "Button1",
            },
          ],
          "node": "element",
          "tag": "button",
        },
        Object {
          "node": "text",
          "text": "",
        },
        Object {
          "attr": Object {
            "class": "uk-button",
            "type": "button",
          },
          "child": Array [
            Object {
              "node": "text",
              "text": "Button2",
            },
          ],
          "node": "element",
          "tag": "button",
        },
        Object {
          "node": "text",
          "text": "",
        },
        Object {
          "attr": Object {
            "class": Array [
              "uk-button",
              "uk-active",
            ],
            "type": "button",
          },
          "child": Array [
            Object {
              "node": "text",
              "text": "Button3",
            },
          ],
          "node": "element",
          "tag": "button",
        },
      ],
      "node": "element",
      "tag": "div",
    },
  ],
  "node": "root",
}
`;
